service/history/historyEngine.go

```diff
@@ -351,7 +351,7 @@ Update_History_Loop:
 		// Start a timer for the decision task.
 		timeOutTask := context.tBuilder.AddDecisionTimoutTask(scheduleID, di.DecisionTimeout)
 		timerTasks := []persistence.Task{timeOutTask}
-		defer e.timerProcessor.NotifyNewTimer(timeOutTask.GetTaskID())
+		defer e.timerProcessor.NotifyNewTimer(timerTasks)
```

Comment:
What happens if the transaction fails? I think we should notify the timer processor only after the transaction succeeds.